gandalf.js

```diff
@@ -1,117 +1,121 @@
-const truncatedString = string.substring(0, maxLength) + (string.length > maxLength ? "..." : "");
```

Comment:
SCRIPT-This function might benefit from using async/await instead of callbacks for better readability.